triciaswift/user-management
# edit user

# Edit User Feature

## Overview

This PR introduces the `EditUser` and `UserForm` components, allowing users to edit an existing user's information. The `EditUser` component is responsible for loading the user data, managing state, and handling the form submission. The `UserForm` component is a presentational component that displays the form inputs for user data.

## Changes

### `EditUser`

- Created a component to handle user data fetching and submission to the server.
- Utilized `useState` for managing user data, loading state, and error messages.
- Integrated `useParams` to retrieve the `userId` from the URL.
- Used `useNavigate` for redirecting after a successful update.
- Added error handling to display messages to the user upon failed requests.
- Implemented a temporary navigation delay to show a success message before redirecting.

### `UserForm`

- Created a form component that takes initial user values and a save handler as props.
- Managed form input changes with local component state.
- Passed down input change handling from the parent `EditUser` component.
- Included form validation to ensure data integrity before submission.

### Error Handling

- Improved user feedback by introducing error state management within the `EditUser` component.
- Errors from API calls are caught and displayed to the user.

### Success Message

- A success message is now displayed upon successful form submission.
- The form is replaced by the success message, preventing further user input and indicating completion of the operation.

## How to Test

1. Navigate to the edit user route (`/users/:userId/edit`).
2. Modify the user's details in the form and submit.
3. Ensure a success message appears and the form disappears.
4. After a brief delay, the application should redirect to the home page.
